tests/ops/functions/test_iterative_qpe.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the iterative_qpe function
+"""
+import pytest
+
+import pennylane as qml
+import numpy as np
+import jax
+
+
+class TestResult:
```

Comment:
We should probably test that it works with all of the measurement processes (expval, probs, etc.) that are supported by mid-circuit measurements.